public/app/features/dashboard/components/SaveDashboard/SaveDashboardDrawer.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { render, waitFor, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import { configureStore } from 'app/store/configureStore';
+
+import { DashboardModel } from '../../state';
+
+import { SaveDashboardDrawer } from './SaveDashboardDrawer';
+
+jest.mock('app/core/core', () => ({
+  ...jest.requireActual('app/core/core'),
+  contextSrv: {},
+}));
+
+jest.mock('@grafana/runtime', () => ({
+  ...jest.requireActual('@grafana/runtime'),
+  getBackendSrv: () => ({
+    post: mockPost,
+  }),
+}));
+
+jest.mock('app/core/services/backend_srv', () => ({
+  backendSrv: {
+    getDashboardByUid: jest.fn().mockResolvedValue({ dashboard: {} }),
+  },
+}));
+const mockOnDismiss = jest.fn();
+const mockPost = jest.fn();
+
+const setup = async () => {
+  const mockDashboard = new DashboardModel({
+    uid: 'mockDashboardUid',
+    version: 1,
+  });
+  const store = configureStore();
+
+  const { rerender } = await waitFor(() =>
+    render(
+      <Provider store={store}>
+        <SaveDashboardDrawer dashboard={mockDashboard} onDismiss={mockOnDismiss} />{' '}
+      </Provider>
+    )
+  );
+
+  return { rerender, mockDashboard, store };
```

Comment:
`setup()` is doing too many things: creating a dashboard mock and returning it, configuring store and returning it, and rendering a component and returning it. I think it would be better to have `mockDashboard` as a part of a single `describe()` or even inside a single test whenever it's needed.
The configuration of the store can also just happen as a separate thing once for the whole test module.